Handle Send button error when no image is selected

xram64 opened the pull request:
Add additional check for NoneType in `image_from_url_text` to catch error caused by clicking "Send" buttons with no image currently selected in the gallery (``AttributeError: 'NoneType' object has no attribute 'startswith'``).

CookieHCl committed:
Make style configurable in ui-config.json

CookieHCl committed:
Use default value when dropdown ui setting is bad

Default value is the first value of selectables.
Particually, None in styles.

CookieHCl committed:
Warn when user uses bad ui setting

xram64 committed:
Handle Send button error when no image is selected

Add additional check for NoneType in `image_from_url_text` to catch error caused by clicking "Send" buttons with no image currently selected in the gallery (``AttributeError: 'NoneType' object has no attribute 'startswith'``).

xram64 committed:
Handle Send button error when no image is selected

Add additional check for NoneType in `image_from_url_text` to catch error caused by clicking "Send" buttons with no image currently selected in the gallery (``AttributeError: 'NoneType' object has no attribute 'startswith'``).

ddPn08 committed:
Allow specifying the region of ngrok.

DancingSnow0517 committed:
fix dir_path in some path like `D:/Pic/outputs`

junpengqiu committed:
Added Refresh Button to embedding and hypernetwork names in Train Tab

Problem
everytime I modified pt files in embedding_dir or hypernetwork_dir, I
need to restart webui to have the new files shown in the dropdown of
Train Tab

Solution
refactored create_refresh_button out of create_setting_component so we
can use this method to create button next to gr.Dropdowns of embedding
name and hypernetworks

Extra Modification
hypernetwork pt are now sorted in alphabetic order

winterspringsummer committed:
added extras batch work from directory

AUTOMATIC1111 committed:
do not reload embeddings from disk when doing textual inversion

AUTOMATIC1111 committed:
honor --hide-ui-dir-config option for #2807

AUTOMATIC1111 commented:
does this improve anything

zeithrold committed:
Pull out some of URL to Env Variable (#2578)

* moved repository url to changeable environment variable

* move stable diffusion repo itself to env

* added missing env

* Remove default URL

Co-authored-by: AUTOMATIC1111 <16777216c@gmail.com>

AUTOMATIC1111 committed:
revert changes to two bat files I asked to revert but the author couldn't in 863e9efc19d2811f1db5055be8e346781df3f7ce.

CookieHCl committed:
Fix FileNotFoundError in history tab

Now only traverse images when directory exists

CookieHCl committed:
Only make output dir when creating output

CookieHCl committed:
Better readablity of logs

AUTOMATIC1111 committed:
keep focus on current element when updating gallery

dvsilch committed:
fix: add null check when start running project the currentButton is null

sgkoishi committed:
The hide_ui_dir_config flag also restrict write attempt to path settings

vacui-dev committed:
Safeguard setting restore logic against exceptions

also useful for keeping settings cache and restore logic together, and nice for code reuse (other third party scripts can import this class)

dfaker committed:
print list of embeddings on reload

DenkingOfficial committed:
Fix CLIP Interrogator and disable ranks for it

MrCheeze committed:
improve performance of 3-way merge on machines with not enough ram, by only accessing two of the models at a time

DancingSnow0517 committed:
add cache for workflow

fortypercnt committed:
Fix #2750

left / top alignment was necessary with gradio 3.4.1. In gradio 3.5 the parent div of the image mask is centered, so the left / top alignment put the mask in the wrong place as described in #2750 #2795 #2805. This fix was tested on Windows 10 / Chrome.

AUTOMATIC1111 committed:
only read files with .py extension from  the scripts dir

AUTOMATIC1111 committed:
fix bug for latest model merge RAM improvement

AUTOMATIC1111 committed:
make CLIP interrogate ranks output sane values

AUTOMATIC1111 committed:
possibly defeat losing of focus for prompt when generating images with gallery open

AUTOMATIC1111 committed:
make modelmerger work with ui-config.json

AUTOMATIC1111 committed:
localization support

xram64 commented:
I guess it's a marginal UX improvement? I just figure a handled exception is better than an unhandled one for something that can come up in normal usage, but it's your call.

Jord-JD committed:
Fix typo in 'choices' when loading upscaler 2 config

dfaker commented:
in this case filedata is `None`, there's only one `None` no need for type comparisons.

`if filedata is None:`

xram64 committed:
Handle Send button error when no image is selected

Add additional check for type `None` in `image_from_url_text` to catch error caused by clicking "Send" buttons with no image currently selected in the gallery (`AttributeError: 'NoneType' object has no attribute 'startswith'`).

xram64 committed:
Merge branch 'dev' of https://github.com/xram64/stable-diffusion-webui into dev

xram64 committed:
Revert "Merge branch 'dev' of https://github.com/xram64/stable-diffusion-webui into dev"

This reverts commit caa78abc99d91f9a93182caa52041398ef20275a, reversing
changes made to 1dab8d48edd4e67cbbeb4e1ebfd2bebeac3fc7fc.

xram64 commented:
Ah ok, cool, I wasn't sure since the error mentioned NoneType.

I tried to make the change but I'm new with merges and it looks like I might have messed up the PR lol
If the change is wanted and it's possible to fix this PR or if I should just make a new one, lmk

dfaker commented:
It's a bit of an edge case, but I think the severity of the "ERROR" appearing in 100 places warrants it.

I think this branch is fucked further than warrants a rebase for this size change, I'll close it, feel free to offer it again cleanly.
